sway-lib-std/src/vec.sw

```diff
@@ -141,4 +142,71 @@ impl<T> Vec<T> {
     pub fn is_empty(self) -> bool {
         self.len == 0
     }
+
+    /// Removes and returns the element at position `index` within the vector,
+    /// shifting all elements after it to the left.
+    /// Panics if `index >= self.len`
+    pub fn remove(mut self, index: u64) -> T {
+        assert(index < self.len);
+
+        let val_size = size_of::<T>();
+        let buf_start = self.buf.ptr();
+        let mut ptr = buf_start + val_size * index;
+
+        // Read from `ptr`
+        let ret = read(ptr);
+
+        // Shift everything down to fill in that spot.
+        let end = buf_start + val_size * self.len;
+        while ptr < end {
+            copy(ptr, ptr + val_size, val_size);
+            ptr += val_size;
+        }
```

Comment:
The loop is needed, can't use `copy(ptr, ptr + val_size, val_size * (self.len - index - 1)` , https://github.com/FuelLabs/sway/issues/2155 discusses this